dmutils/filters.py

```diff
@@ -24,19 +25,23 @@ def format_links(text):
                                 \.                        # match dot in host name
                                 (?:[^\s/?\.#<>"']+\.?)+   # match rest of domain
                                 (?:/[^\s<>"']*)?          # rest of url
-                                [^\s<>"'\.]               # no dot at end
+                                [^\s<>,"'\.]              # no dot at end
                                 )""", re.X)
-    matches = url_match.findall(text)
-    link = '{0}<a href={1} rel="external">{1}</a>'
-    if matches:
-        textArray = url_match.split(text)
-        formattedText = ""
-        for text in textArray:
-            if text in matches:
-                formattedText = link.format(formattedText, Markup.escape(text))
+    matched_urls = url_match.findall(text)
+    if matched_urls:
+        link = '<a href="{0}" rel="external">{1}</a>'
+        formatted_text = ""
+        text_array = url_match.split(text)
+        formatted_text_array = []
+        for partial_text in text_array:
+            if partial_text in matched_urls:
+                url = "https://{}".format(partial_text) if partial_text.startswith('www') else partial_text
```

Comment:
There's no warning when connecting to a web server that doesn't accept HTTPS requests at all (which feels like a more common scenario than invalid certificate/mixed content) so it will only say "This site can't be reached". `//` is relative to the origin, which means that since Digital Marketplace is https it will make all links https. 

I don't think there's a good answer to this. We're not going to build a link checker to test whether the linked page is accessible via HTTP or HTTPS, so this means we will end up with broken links in both cases (probably more if we go with `https://`). We could look at links in the existing briefs to get more data about this, but my suggestion is still to not link `www.` and let suppliers copy the link manually. This way we don't create an expectation that the link is correct and the website isn't working for some reason and might prompt the user to try a different approach (eg changing http to https manually or pasting the url to a search engine).